spring-pulsar-spring-boot-autoconfigure/src/main/java/org/springframework/pulsar/autoconfigure/PulsarProperties.java

```diff
@@ -1261,7 +1277,7 @@ public Map<String, Object> buildProperties() {
 			map.from(this::getListenerName).to(properties.in("listenerName"));
 			map.from(this::getAuthPluginClassName).to(properties.in("authPluginClassName"));
 			map.from(this::getAuthParams).to(properties.in("authParams"));
-			map.from(this::getAuthParamsMap).to(properties.in("authParamMap"));
+			map.from(convertToEncodedParamString(this.getAuthParamsMap())).to(properties.in("authParams"));
```

Comment:
I was also thinking in the case where the user has set `authParams` directly that they are wanting to be in control and in that case backoff the special processing.  I think as you have it currently they will be overwritten.  Wdyt?


_NOTE: Only pasting this because I had it drafted locally.. not saying "use my code instead"._ 

```java
// If authParams is set, leave as-is, user is in control.
if (StringUtils.hasText(this.getAuthParams())) {
  map.from(this::getAuthParams).to(properties.in("authParams"));
  map.from(this::getAuthParamsMap).to(properties.in("authParamMap"));
}
else if (!CollectionUtils.isEmpty(this.getAuthParamsMap())) { // Otherwise, build authParams from authParamsMap
  try {
    String authParamsJson = ObjectMapperFactory.create().writeValueAsString(this.authParamsMap);
    map.from(authParamsJson).to(properties.in("authParams"));
  }
  catch (JsonProcessingException e) {
    throw new IllegalStateException("Failed to convert authParamsMap to JSON string", e);
  }
}

```

Reply:
AFAIK the client code configures the Authentication with either `authParams` or `authParamsMap`, not both (I have to confirm this though). If that is the case, I think the current way of using the parameter map in case it is not empty seems fine to me.

Reply:
If that is the case then, agreed. I think we should rename from `convert...` to `maybeConvert...` to indicate that it will do nothing in the empty case a bit more clearly. Nit though.

Reply:
Just confirmed it, the client implementation will exclusively use the String authParams when set, or fall back to the map: https://github.com/apache/pulsar/blob/master/pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java#L253-L257

Reply:
Good to know @alpreu . 

However, now that we have renamed from `authParamsMap -> authentication` I think this situation can now occur. User sets some `authParams` and then also some `authentication` params,, we will overwrite the former. I wonder if we should throw an exception if the user sets `authParams | authParamsMap` stating they should use `authentication` instead.

Reply:
Ah I see, good point. Yes, let's throw an exception if both are set